drivers/hmis/app/graphql/mutations/join_household.rb

```diff
@@ -0,0 +1,103 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class JoinHousehold < BaseMutation
+    argument :receiving_household_id, ID, required: true
+    argument :joining_enrollment_inputs, [Types::HmisSchema::EnrollmentRelationshipInput], required: true
+
+    field :receiving_household, Types::HmisSchema::Household, null: false
+    field :donor_household, Types::HmisSchema::Household, null: true # Will be null if there are no remaining members
+    # todo @martha - possible that there will be multiple households with same ID? mutation should require project_id?
+    def resolve(receiving_household_id:, joining_enrollment_inputs:)
+      receiving_enrollments = Hmis::Hud::Enrollment.
+        where(household_id: receiving_household_id).
+        viewable_by(current_user).
+        includes(:project, :household)
+      access_denied! if receiving_enrollments.empty?
+
+      receiving_household = receiving_enrollments.first.household
+
+      receiving_project = receiving_enrollments.map(&:project).uniq.sole
```

Comment:
I think this could be simplified by using the household view directly (untested)
```suggestion
      # note: viewable_by takes care of filtering by data source via project with_access
      receiving_household = Hmis::Hud::Household.viewable_by(current_user).find_by(household_id: receiving_household_id)
      access_denied! unless receiving_household

      receiving_enrollments = receiving_household.enrollments
      receiving_project = receiving_household.project
```